sql/updates/2.14.2--2.15.0.sql

```diff
@@ -89,23 +89,36 @@ FROM _timescaledb_catalog.hypertable;
 -- Rebuild the catalog table `_timescaledb_catalog.continuous_aggs_bucket_function`
 --
 
-CREATE OR REPLACE FUNCTION _timescaledb_functions.cagg_get_bucket_function(
-    mat_hypertable_id INTEGER
-) RETURNS regprocedure AS '@MODULE_PATHNAME@', 'ts_continuous_agg_get_bucket_function' LANGUAGE C STRICT VOLATILE;
+CREATE OR REPLACE FUNCTION _timescaledb_functions.cagg_get_bucket_function_info(
+    mat_hypertable_id INTEGER,
+    -- The bucket function
+    OUT bucket_func REGPROCEDURE,
+    -- `bucket_width` argument of the function, e.g. "1 month"
+    OUT bucket_width TEXT,
+    -- optional `origin` argument of the function provided by the user
+    OUT bucket_origin TEXT,
+    -- optional `offset` argument of the function provided by the user
+    OUT bucket_offset TEXT,
+    -- optional `timezone` argument of the function provided by the user
+    OUT bucket_timezone TEXT,
+    -- fixed or variable sized bucket
+    OUT bucket_fixed_width BOOLEAN
+) RETURNS RECORD AS '@MODULE_PATHNAME@', 'ts_continuous_agg_get_bucket_function_info' LANGUAGE C STRICT VOLATILE;
```

Comment:
linking to C functions is problematic especially when this are removed as that will create a dependency on .so file

Reply:
This is a new function introduced by #7042.  The other function (`cagg_get_bucket_function`) that was "temporary" used only for update purposes was removed from the code base but it was never exposed for the users.

Reply:
Reverted the code!